File: 17. Dynamic array allocation.cpp
// Topic: Dynamic array allocation using new, fill and print
#include <iostream>
using namespace std;

int main()
{
    int n = 5;
    int* arr = new int[n];
    for (int i = 0; i < n; ++i) arr[i] = i * 2;
    for (int i = 0; i < n; ++i) cout << arr[i] << ' ';
    cout << '\n';
    delete[] arr;
    return 0;
}

// Time Complexity: O(n) to fill and print.
// Space Complexity: O(n) for dynamically allocated array.

File: 17. Dynamic array allocation.py
"""
Topic: Dynamic array allocation (Python list demo)
"""
def main():
    n = 5
    arr = [0]*n
    for i in range(n): arr[i] = i*2
    print(arr)

if __name__ == '__main__':
    main()

# Time Complexity: O(n).
# Space Complexity: O(n).
